api_v3/lib/KalturaLiveEntryService.php

```diff
@@ -246,26 +246,40 @@ function registerMediaServerAction($entryId, $hostname, $mediaServerIndex, $appl
 		}
 
 		$dbEntry->setRedirectEntryId(null);
-		
 		if($dbEntry->save())
 		{
 			if($mediaServerIndex == MediaServerIndex::PRIMARY && $dbEntry->getRecordStatus())
 			{
+				KalturaLog::info("Checking if recorded entry needs to be created for entry $entryId");
 				$createRecordedEntry = false;
 				if(!$dbEntry->getRecordedEntryId())
 				{
 					$createRecordedEntry = true;
+					KalturaLog::info("Creating new recoreded entry for $entryId ");
 				}
-				elseif($dbEntry->getRecordStatus() == RecordStatus::PER_SESSION && ($dbEntry->getLastBroadcastEndTime() + kConf::get('live_session_reconnect_timeout', 'local', 180) < time()))
-				{
-					$createRecordedEntry = true;
+				else {
+					$dbRecordedEntry = entryPeer::retrieveByPK($dbEntry->getRecordedEntryId());
+					if (!$dbRecordedEntry) {
+						throw new KalturaAPIException(KalturaErrors::ENTRY_ID_NOT_FOUND, $dbRecordedEntry);
+					}
+					$recordedEntryCreationTime = strtotime($dbRecordedEntry->getCreatedAt());
+
+					$isNewSession = $dbEntry->getLastBroadcastEndTime() + kConf::get('live_session_reconnect_timeout', 'local', 180) < $dbEntry->getCurrentBroadcastStartTime();
+					$recordedEntryNotYetCreatedForCurrentSession = $recordedEntryCreationTime < $dbEntry->getCurrentBroadcastStartTime();
+
+					if ($dbEntry->getRecordStatus() == RecordStatus::PER_SESSION) {
+						if ($isNewSession && $recordedEntryNotYetCreatedForCurrentSession)
```

Comment:
AND.
In order to create new recored entry, one require that
1. The session is new, namely, more than 3 minutes were passed from the last broadcast time.
2. Since, the current function (registerMediaServerAction) might called every one minutes, we also need to ensure that it the first time that we create new recorded entry.